src/buildOptionsList.js

```diff
@@ -4,13 +4,23 @@ import Icon from "react-native-vector-icons/AntDesign";
 
 import getDeviceInfo from "./getDeviceInfo";
 
-export const buildList = (items, setVisible, mail, reportIcon, dismissIcon) => {
+export const buildOptionsList = (
+  items,
+  setVisible,
+  email,
+  reportIcon,
+  dismissIcon,
+  reportIssue,
+) => {
   const handleReport = () => {
     const info = getDeviceInfo();
-    if (mail === undefined) {
-      alert("You didn't provide an email address as prop")
+
+    if (reportIssue !== undefined) {
+      reportIssue?.(info);
+    } else if (email !== undefined) {
```

Comment:
Again, we decided to allow both methods of reporting the issue. So they shouldn't be linked with that `else`.
```
if (typeof reportIssue === 'function') { // here you should throw an error when it is not a function
   reportIssue?.(info);
} 

 if (typeof email !== 'string') { // if the email is not a string we should throw an error
    // send the email
}
```